oai-pmh-handler/src/main/java/no/sikt/nva/oai/pmh/handler/FormUrlencodedBodyParser.java

```diff
@@ -0,0 +1,66 @@
+package no.sikt.nva.oai.pmh.handler;
+
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import nva.commons.core.StringUtils;
+
+public final class FormUrlencodedBodyParser {
+
+  private static final String AMPERSAND = "&";
+  private static final String EQUALS_SIGN = "=";
+  private static final int SPLIT_PARTS_LIMIT = 2;
+  private static final int ONE = 1;
+
+  private Map<String, String> parameters;
+
+  private FormUrlencodedBodyParser(String body) {
+    if (StringUtils.isEmpty(body)) {
+      throw new IllegalArgumentException("Body cannot be null or empty");
+    }
```

Comment:
This will never be triggered given the code that precedes it, right?

Reply:
Right, but hopefully this can be used elsewhere, so the class should be defensive ;)